bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/HumiditySensorHandler.java

```diff
@@ -0,0 +1,144 @@
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        final OmnilinkBridgeHandler bridgeHandler = getOmnilinkBridgeHandler();
+        if (bridgeHandler != null) {
+            updateHumiditySensorProperties(bridgeHandler);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Received null bridge while initializing Humidity Sensor!");
+        }
+    }
+
+    private void updateHumiditySensorProperties(OmnilinkBridgeHandler bridgeHandler) {
+        final List<AreaProperties> areas = getAreaProperties();
+        if (areas != null) {
+            for (AreaProperties areaProperties : areas) {
+                int areaFilter = bitFilterForArea(areaProperties);
+
+                ObjectPropertyRequest<AuxSensorProperties> objectPropertyRequest = ObjectPropertyRequest
+                        .builder(bridgeHandler, ObjectPropertyRequests.AUX_SENSORS, thingID, 0).selectNamed()
+                        .areaFilter(areaFilter).build();
+
+                for (AuxSensorProperties auxSensorProperties : objectPropertyRequest) {
+                    Map<String, String> properties = editProperties();
+                    properties.put(THING_PROPERTIES_NAME, auxSensorProperties.getName());
+                    properties.put(THING_PROPERTIES_AREA, Integer.toString(areaProperties.getNumber()));
+                    updateProperties(properties);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand called for channel: {}, command: {}", channelUID.getAsString(), command);
+
+        if (!(command instanceof DecimalType)) {
+            logger.warn("Invalid command: {}, must be DecimalType", command);
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_AUX_LOW_SETPOINT:
+                sendOmnilinkCommand(OmniLinkCmd.CMD_THERMO_SET_HEAT_LOW_POINT.getNumber(),
+                        TemperatureFormat.FAHRENHEIT.formatToOmni(((DecimalType) command).intValue()), thingID);
+                break;
+            case CHANNEL_AUX_HIGH_SETPOINT:
+                sendOmnilinkCommand(OmniLinkCmd.CMD_THERMO_SET_COOL_HIGH_POINT.getNumber(),
+                        TemperatureFormat.FAHRENHEIT.formatToOmni(((DecimalType) command).intValue()), thingID);
+                break;
+            default:
+                logger.warn("Unknown channel for Humdity Sensor thing: {}", channelUID.getAsString());
+        }
+    }
+
+    @Override
+    public void updateChannels(ExtendedAuxSensorStatus status) {
+        logger.debug("updateChannels called for Humidity Sensor status: {}", status);
+        updateState(CHANNEL_AUX_HUMIDITY,
+                new DecimalType(TemperatureFormat.FAHRENHEIT.omniToFormat(status.getTemperature())));
```

Comment:
Because it already used DecimalType in the previous version of this binding and I did not think to change it, unless you think it is important, I would prefer to leave it as is.

Reply:
using a quantity type is simple, `new QuantityType<>(temp, temperatureUnit);` where temp is your float value and temperatureUnit is either `org.openhab.core.library.unit.ImperialUnits.FAHRENHEIT` or `org.openhab.core.library.unit.SIUnits.CELSIUS` depending on what the omni is reporting the value in

Reply:
I'll work on implementing this, what do you use for humidity with QuantityType

Reply:
@J-N-K Does that look better?